rotkehlchen/types.py

```diff
@@ -467,3 +467,27 @@ class ExchangeApiCredentials(NamedTuple):
     Location.BISQ,
 ]
 EXTERNAL_LOCATION = [Location.EXTERNAL] + EXTERNAL_EXCHANGES
+
+
+class ExchangeLocationID:
+    def __init__(self, name: str, location: Location) -> None:
+        self.name = name
+        self.location = location
+
+    def __str__(self) -> str:
+        return f'{self.name}-{self.location.name}'
+
+    def __hash__(self) -> int:
+        return hash(str(self))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, ExchangeLocationID):
+            return NotImplemented
+        return self.name == other.name and self.location == other.location
+
+    def serialize(self) -> Dict:
+        return {'name': self.name, 'location': self.location.value}
+
+    @staticmethod
+    def deserialize(data: Dict) -> 'ExchangeLocationID':
+        return ExchangeLocationID(data['name'], Location(data['location']))
```

Comment:
Here it shouldn't happen but is best to protect agains possible errors. Location could fail to find the location and you should catch it and raise `DeserializationError`

Reply:
Shall I also check for example whether required keys (name, location) exist in the dict or it is redundant?

Reply:
@yabirgb what do you think about it?